firebase-firestore/src/test/java/com/google/firebase/firestore/model/FieldIndexTest.java

```diff
@@ -88,4 +91,26 @@ public void comparatorIncludesSegmentsLength() {
     assertEquals(0, SEMANTIC_COMPARATOR.compare(indexOriginal, indexSame));
     assertEquals(-1, SEMANTIC_COMPARATOR.compare(indexOriginal, indexDifferent));
   }
+
+  @Test
+  public void indexOffsetComparator() {
+    IndexOffset docAOffset = IndexOffset.create(version(1), key("foo/a"));
+    IndexOffset docBOffset = IndexOffset.create(version(1), key("foo/b"));
+    IndexOffset version1Offset = IndexOffset.create(version(1));
+    IndexOffset docCOffset = IndexOffset.create(version(2), key("foo/c"));
+    IndexOffset version2Offset = IndexOffset.create(version(2));
+
+    assertEquals(-1, docAOffset.compareTo(docBOffset));
+    assertEquals(-1, docAOffset.compareTo(version1Offset));
+    assertEquals(-1, version1Offset.compareTo(docCOffset));
+    assertEquals(-1, version1Offset.compareTo(version2Offset));
+    assertEquals(-1, docCOffset.compareTo(version2Offset));
+  }
+
+  @Test
+  public void indexOffsetAdvancesSeconds() {
+    IndexOffset actualSuccessor = IndexOffset.create(version(1, (int) 1e9 - 1));
+    IndexOffset expectedSuccessor = IndexOffset.create(version(2, 0), DocumentKey.empty());
```

Comment:
ultranit: no need to specify DocumentKey.empty()

Reply:
We do need to specify DocumentKey.empty() or else we create a key with `version(3,0), DocumentKey.empty()`.